datasets/program_executor.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC
+
+from datasets.context import Context
+
+
+class ProgramExecutor(ABC):
+    """
+    Class to access information about the ML program currently being executed.
```

Comment:
I've noticed a couple of references to ML specifically throughout some comments here, I'm working if the `datasets` library can be thought of more generically than that?

Reply:
True, it could be thought of as consistent logical data access (like [intake-dal](https://github.com/zillow/intake-dal/))